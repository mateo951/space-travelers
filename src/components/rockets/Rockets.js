import React from 'react';

const Rockets = () => (
  <h2>Rockets</h2>
);

export default Rockets;
