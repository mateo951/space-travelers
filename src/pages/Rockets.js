import React from 'react';
import RocketList from '../components/rockets/RocketList';

const Rockets = () => (
  <>
    <div className="container">
      <div className="inner-container">
        <RocketList />
      </div>
    </div>
  </>
);

export default Rockets;
