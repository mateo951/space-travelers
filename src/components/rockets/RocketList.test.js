import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import store from '../../redux/configureStore';
import Rockets from '../../pages/RocketsPage';

describe('Rockets component', () => {
  describe('Use Jest snapshots to test the Rockets component', () => {
    it('renders Rockets component as expected', async () => {
      const RocketsComponent = render(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      );

      expect(await screen.findAllByText('Reserve Rocket')).toBeTruthy();
      expect(RocketsComponent.asFragment()).toMatchSnapshot();
    });
  });
});
