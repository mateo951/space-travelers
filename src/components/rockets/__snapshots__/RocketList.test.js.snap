// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component Use Jest snapshots to test the Rockets component renders Rockets component as expected 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="inner-container"
    >
      <ul
        class="rocketsList"
      >
        <li
          class="rocketItem"
        >
          <img
            alt="rocket"
            class="rocketImage"
            src="https://imgur.com/DaCfMsj.jpg,https://imgur.com/azYafd8.jpg"
          />
          <div
            class="descContainer"
          >
            <h3>
              Falcon 1
            </h3>
            <p>
               The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
            </p>
            <div>
              <button
                class="submitReserve"
                id="1"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </li>
        <li
          class="rocketItem"
        >
          <img
            alt="rocket"
            class="rocketImage"
            src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg,https://farm4.staticflickr.com/3955/32915197674_eee74d81bb_b.jpg,https://farm1.staticflickr.com/293/32312415025_6841e30bf1_b.jpg,https://farm1.staticflickr.com/623/23660653516_5b6cb301d1_b.jpg,https://farm6.staticflickr.com/5518/31579784413_d853331601_b.jpg,https://farm1.staticflickr.com/745/32394687645_a9c54a34ef_b.jpg"
          />
          <div
            class="descContainer"
          >
            <h3>
              Falcon 9
            </h3>
            <p>
               Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.
            </p>
            <div>
              <button
                class="submitReserve"
                id="2"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </li>
        <li
          class="rocketItem"
        >
          <img
            alt="rocket"
            class="rocketImage"
            src="https://farm5.staticflickr.com/4599/38583829295_581f34dd84_b.jpg,https://farm5.staticflickr.com/4645/38583830575_3f0f7215e6_b.jpg,https://farm5.staticflickr.com/4696/40126460511_b15bf84c85_b.jpg,https://farm5.staticflickr.com/4711/40126461411_aabc643fd8_b.jpg"
          />
          <div
            class="descContainer"
          >
            <h3>
              Falcon Heavy
            </h3>
            <p>
               With the ability to lift into orbit over 54 metric tons (119,000 lb)--a mass equivalent to a 737 jetliner loaded with passengers, crew, luggage and fuel--Falcon Heavy can lift more than twice the payload of the next closest operational vehicle, the Delta IV Heavy, at one-third the cost.
            </p>
            <div>
              <button
                class="submitReserve"
                id="3"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </li>
        <li
          class="rocketItem"
        >
          <img
            alt="rocket"
            class="rocketImage"
            src="https://live.staticflickr.com/65535/48954138962_ee541e6755_b.jpg,https://live.staticflickr.com/65535/48953946911_e60c5bcc5c_b.jpg,https://live.staticflickr.com/65535/48954138922_9c42173f08_b.jpg,https://live.staticflickr.com/65535/48953947006_313f01ec93_b.jpg"
          />
          <div
            class="descContainer"
          >
            <h3>
              Starship
            </h3>
            <p>
               Starship and Super Heavy Rocket represent a fully reusable transportation system designed to service all Earth orbit needs as well as the Moon and Mars. This two-stage vehicle — composed of the Super Heavy rocket (booster) and Starship (ship) — will eventually replace Falcon 9, Falcon Heavy and Dragon.
            </p>
            <div>
              <button
                class="submitReserve"
                id="4"
                type="button"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
